wikimedia/wikipedia-ios
# T287970 mark as seen

**Phabricator:** https://phabricator.wikimedia.org/T287970

### Notes
* This PR adds the capability to mark notifications as seen from the app, replicating them to the web. As the user enters the notifications view, a call will be made to the notifications API and mark all notifications as seen unless a notifications inbox filter is applied.

### Test Steps
1. Be sure to have unseen notifications on a project. You can check it by going to the web and making sure the badge on the bell icon is red
2. On the iOS App, click on the notifications icon on the upper left corner of the Explore feed. Do not click on the notifications.
3. Go to the desktop version, and check if the badge is now grey.

### Screenshots/Videos
- Badge with unseen notifications
<img width="43" alt="Screen Shot 2022-02-10 at 18 16 13" src="https://user-images.githubusercontent.com/31116510/153499109-5a9fe145-c133-4cd4-970a-5ed71783c2ac.png">

- Badge with seen and unread notifications
<img width="39" alt="Screen Shot 2022-02-10 at 18 16 45" src="https://user-images.githubusercontent.com/31116510/153499215-b305f6c9-b3e3-4fe2-a548-53ccb1eb60f5.png">
